src/systems/camera_movement.rs

```diff
@@ -20,11 +20,15 @@ impl<'s> System<'s> for CameraMovementSystem {
         WriteStorage<'s, Transform>,
         ReadStorage<'s, Camera>,
         Read<'s, InputHandler<StringBindings>>,
-        ReadExpect<'s, ScreenDimensions>,
+        ReadStorage<'s, Player>,
     );
 
-    fn run(&mut self, (mut transforms, cameras, input, _screen): Self::SystemData) {
-        for (_camera, transform) in (&cameras, &mut transforms).join() {
+    fn run(&mut self, (mut transforms, cameras, input, players): Self::SystemData) {
+
+        let player_pos = (&players, &transforms).join().next()
+            .map(|(_p, tf)| tf)
```

Comment:
Personally I prefer readable names like `transform` (the problem with `tf`, that I has to check the surrounding code, to know what is it standing for).
(player is actually omitted here, so it can be acceptible).
To clarify `pos` in `player_pos` is a widely used, generally known abbreviation, so it's totally OK.

Reply:
fixedpushed